### PR TITLE
Automated cherry pick of #94107: Updating kube-proxy to trim space from

### PR DESCRIPTION
**Pull request by robscott**
Cherry pick of #94107 on release-1.19.

#94107: Updating kube-proxy to trim space from

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

File: pkg/proxy/iptables/proxier_test.go [hunk 1]
```diff
@@ -687,6 +687,10 @@ func TestLoadBalancer(t *testing.T) {
 			svc.Status.LoadBalancer.Ingress = []v1.LoadBalancerIngress{{
 				IP: svcLBIP,
 			}}
+			// Also ensure that invalid LoadBalancerSourceRanges will not result
+			// in a crash.
+			svc.Spec.ExternalIPs = []string{svcLBIP}
+			svc.Spec.LoadBalancerSourceRanges = []string{" 1.2.3.4/28"}
```
Comment (bowei) [inline pkg/proxy/iptables/proxier_test.go:693]: // Leading whitespace.

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**